### ISSUE TITLE
W1D4T2 Strange behavior of cross_val_score

### ISSUE THREAD

**Issue by da5nsy** [NeuromatchAcademy/course-content#346]
My understanding: `cross_val_score` should do a number of splits (determined by the value of `cv`).
One would assume that on real data the accuracy for each split would be different.

In the demo case the the two of the outputs are identical. 

```
accuracies = cross_val_score(LogisticRegression(penalty='none'), X, y, cv=8) # k=8 crossvalidation

print(accuracies)
print(accuracies[4] == accuracies[7])

> [0.85714286 0.8        0.88571429 0.94285714 0.79411765 0.94117647 0.91176471 0.79411765]
> True
```

My estimate:
90% chance this is my misunderstanding. 
10% chance this is a bug in scikit-learn / this dataset / the way the argument is called.

**Comment by mwaskom** [comment]
Bear in mind that, with k fold cross validation on a not-especially large dataset, there will be a relatively small number of observations in each validation set (I think about 34 here). So there are a limited number of unique values that an accuracy score (the proportion of correct predictions) can take.

I think that the intuition to suspect a bug when you see exact matches in floating point numbers is generally good, but I am pretty confident it's leading you astray here.

**Comment by da5nsy** [comment]
I suspect you are right @mwaskom! When I have more time after the course I'll dig into this.

**Comment by mwaskom** [comment]
First debugging step would be to iterate through the cross val splits (like in W1D3) and see if it's actually making the exact same predictions on those splits or just making the same number of correct predictions. You can also go backwards through the GLM math to see how it's arriving at those predictions.

Or, theory project: what is the probability of this happening? :)
